### PR TITLE
Linked coupling

### PR DESCRIPTION
**Pull request by JensWehner**

### CODE REVIEW

File: include/votca/xtp/aomatrix.h [hunk 1]
```diff
@@ -31,265 +31,313 @@
...
-    private: 
-        void setAPolarSite(ctp::APolarSite* site){
-            apolarsite=site;
+}
+
+namespace votca {
+    namespace xtp {
+        namespace ub = boost::numeric::ublas;
+
+        /* "superclass" AOSuperMatrix contains all common functionality for
```
Comment (JoshuaSBrown) [inline include/votca/xtp/aomatrix.h]: It would be preferable to reserve the /* comment type for debugging and use // for comments in the code.

Comment (JoshuaSBrown) [reply include/votca/xtp/aomatrix.h]: What is this branch for exactly?

Comment (JensWehner) [reply include/votca/xtp/aomatrix.h]: 1) agreed 
2) Due to some merges, the true purpose of this branch got a little mixed, it mainly focuses on electronic coupling using a linker atom

**Comment by JoshuaSBrown** [comment]
Would also be nice to have a few tests automated to check that everything is running correctly.  Take what you will from my feedback :).

**Comment by junghans** [comment]
Travis fails, because it needs a newer tools version merged into votca/votca. This is done in votca/votca#54.

**Comment by junghans** [comment]
Travis says:
```
/home/votca/votca/xtp/src/libxtp/qmpackage.cc:200:31: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
             for (int i = 0; i < linker_names.size(); i++) {
                             ~~^~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

**Comment by junghans** [comment]
@jens: Is this ready to be merged?

**Comment by JensWehner** [comment]
@junghans  it is ready now. I need a reviewer. Look at eigen.h

**Pull request merged**